src/App.jsx

```diff
@@ -51,7 +64,16 @@ export function App() {
 		<Router>
 			<Routes>
 				<Route path="/" element={<Layout />}>
-					<Route index element={<Home />} />
+					<Route
+						index
+						element={
+							listToken ? (
+								<List data={data} />
+							) : (
+								<Home makeNewList={(token) => setList(token)} />
+							)
+						}
+					/>
```

Comment:
The simple ternary here makes it really clear as to what will render according to the existence of the listToken.